File: ucare_backend/src/main/java/com/douzone/ucare/service/PatientService.java
package com.douzone.ucare.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.douzone.ucare.repository.PatientRepository;
import com.douzone.ucare.vo.PatientVo;
import com.douzone.ucare.vo.UserVo;

@Service
public class PatientService {
	
	@Autowired
	private PatientRepository PatientRepository;

	public int create(PatientVo patient) {
		return PatientRepository.create(patient);
	}

	public List<PatientVo> retrieveAll() {
		return PatientRepository.retrieveAll();
	}
	
	public PatientVo retrieve(Long patientNo) {
		return PatientRepository.retrieve(patientNo);
	}
	
	public int update(PatientVo patient) {
		return PatientRepository.update(patient);
	}
	
	public int updateDiagnosis(PatientVo patient) {
		return PatientRepository.updateDiagnosis(patient);
	}
	
	public int ssnOverlap(PatientVo patient) {
		return PatientRepository.ssnOverlap(patient);
	}
}

File: ucare_frontend/src/service/patientService.js
import axios from 'axios';
import {PATIENT_API_BASE_URL} from './urlConfig'

class patientService {
  
  create(data){
    return axios.post(PATIENT_API_BASE_URL + '/create', data, { headers: { Authorization: localStorage.getItem("authorization") }});
  }

  retrieveAll(){
    return axios.get(PATIENT_API_BASE_URL + '/retrieveAll', { headers: { Authorization: localStorage.getItem("authorization") }});
  }

  retrieve(patientNo){
    return axios.get(PATIENT_API_BASE_URL + '/retrieve/' + patientNo, { headers: { Authorization: localStorage.getItem("authorization") }});
  }

  update(data){
    return axios.put(PATIENT_API_BASE_URL + '/update', data, { headers: { Authorization: localStorage.getItem("authorization") }});
  }

  updateDiagnosis(data){
    return axios.put(PATIENT_API_BASE_URL + '/updateDiagnosis', data, { headers: { Authorization: localStorage.getItem("authorization") }});
  }

  ssnOverlap(data){
    return axios.post(PATIENT_API_BASE_URL + '/ssnOverlap', data, { headers: { Authorization: localStorage.getItem("authorization") }});
  }
}

export default new patientService();